src/utils/query/Query.tsx

```diff
@@ -37,5 +37,13 @@ export const useQuery = <T,>(
     dispatch({ tag: "fetch" });
   }, [dispatch]);
 
+  React.useEffect(() => {
+    document.addEventListener("visibilitychange", () => {
+      if (document.visibilityState === "visible") {
+        refetch();
+      }
+    });
+  }, [refetch]);
```

Comment:
ini event nya perlu di cleanup kalau componentnya di unmount. Biar gak jalan lagi

```suggestion
  React.useEffect(() => {
    const refetchOnVisible = () => {
      if (document.visibilityState === "visible") {
        refetch();
      }
    }
  
    document.addEventListener("visibilitychange", refetchOnVisible);
    
    return () => {
      document. removeEventListener("visibilitychange", refetchOnVisible)
    }
  }, [refetch]);
```

Reply:
Ohh I see, Today  I Learn